### DISCUSSION TITLE
Is the KEDA azure event hub scaler available for node applications using Dapr on Azure Container App?

### DISCUSSION THREAD

**Discussion by ChedyMissaouiAsForvia** [kedacore/keda#6037] [Q&A / Need Help]
After testing the KEDA azure event hub scaler on a container app running a node js application that consumes event hub messages via [dapr input bindings,](https://docs.dapr.io/developing-applications/building-blocks/bindings/bindings-overview/) we noticed that the container app never scales beyond 1 replica.

**Given the following:**

- Event hub topic has 1 partition
- We applied a load in the order of 100000s of messages per second-ish which should be enough to cause the auto scale
- We have several rules configured for several event hub topics
- We have the following which is one of the auto-scalers config *(some values are redacted):*
```tf
minReplicas = 0
maxReplicas = 4
rules = [
    name = "scale-rule-topic-0"
      custom = {
        type     = "azure-eventhub"
        metadata = {
          eventHubNamespace                   = "***"
          eventHubName                        = "***"
          consumerGroup                       = "***"
          storageAccountName                  = "***"
          blobContainer                       = "***"
          activationUnprocessedEventThreshold = "1"
          unprocessedEventThreshold           = "1"
          checkpointStrategy                  = "blobMetadata"
        }
   }
   ...
]
```

We have never observed an auto-scale despite a load that should largely trigger one.

[In the KEDA azure event-hub scaler documentation,](https://keda.sh/docs/2.15/scalers/azure-event-hub/#checkpointing-behaviour) we have signs that the scaler is only supported for .NET and Python with the event-hub SDK. **Can you please confirm this ?**

In addition, [we have read mentioned in this discussion](https://github.com/microsoft/azure-container-apps/issues/872#issuecomment-1694245078) that max number of replicas is equal to number of partitions in eventhub. **Can you please confirm this also ?**

**Comment by tomkerkhove** [comment]
> In addition, [we have read mentioned in this discussion](https://github.com/microsoft/azure-container-apps/issues/872#issuecomment-1694245078) that max number of replicas is equal to number of partitions in eventhub. Can you please confirm this also ?

Correct.

However, for support on Azure Container Apps we recommend reaching out to them given we are not in charge of that

**Reply by ChedyMissaouiAsForvia** [reply]
Thank you @tomkerkhove for your confirmation.
